### PR TITLE
Add new SSL input plugin

### PR DESCRIPTION
**Pull request by mgrabazey**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

A new simple plug-in for checking the time to expiration of the ssl certificate and its correspondence to the domain

### CODE REVIEW

**Comment by danielnelson** [comment]
@mgrabazey Can you take a look at this similar pull request and let me know if it would fulfill your use? #3768

**Comment by mgrabazey** [comment]
This implementation does not check whether the certificate matches the domain on which the certificate is used

File: plugins/inputs/ssl/ssl.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+func getServerCertsChain(d string, p string, t int) ([]*x509.Certificate, error) {
+	h := getServerAddress(d, p)
+	ipConn, err := net.DialTimeout("tcp", h, time.Duration(t)*time.Second)
+	if err != nil {
+		return nil, errors.New("[" + h + "] " + err.Error())
+	}
+	defer ipConn.Close()
+
+	tlsConn := tls.Client(ipConn, &tls.Config{ServerName: d, InsecureSkipVerify: true})
```
Comment (phemmer) [inline plugins/inputs/ssl/ssl.go]: If we're going to set `InsecureSkipVerify: true`, we should probably do a manual verification step and indicate the status. It's possible for an intermediate or root cert to be revoked (or expired in the case of a stupid CA), which would invalidate the cert.

**Comment by danielnelson** [comment]
@mgrabazey We did merge the alternate plugin for certificate checking, and I think we should be able to integrate the certificate verification pieces into that plugin.  I opened a feature request for this addition #4877.

**Pull request closed without merging**